### ISSUE TITLE
`Style/RedundantRegexpCharacterClass` detects no offenses

### ISSUE THREAD

**Issue by nobuyo** [rubocop/rubocop#11651]
[This testcase](https://github.com/rubocop/rubocop/blob/master/spec/rubocop/cop/style/redundant_regexp_character_class_spec.rb#L310-L326) fails on my local machine (M2 macbook air) while developing https://github.com/rubocop/rubocop/pull/11644.

--------

## Expected behavior

`Style/RedundantRegexpCharacterClass` add offense to `/[\032]/`.

## Actual behavior

No offenses detected.

## Steps to reproduce the problem

```rb
# Gemfile

# frozen_string_literal: true

source 'https://rubygems.org'

gem 'rubocop'
```

```yml
# .rubocop.yml
AllCops:
  NewCops: enable

Style/RedundantRegexpCharacterClass:
  Enabled: true
```

```rb
# test.rb

# frozen_string_literal: true

/[\032]/
```

Run `bundle exec rubocop --only Style/RedundantRegexpCharacterClass test.rb` and got result:

```console
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

on rubocop repository:
```
$ [bundle exec] rubocop -V
1.47.0 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 3.2.0) [arm64-darwin21]
  - rubocop-performance 1.16.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.18.1
```

on small reproduction code:
```
$ [bundle exec] rubocop -V
1.47.0 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 3.2.0) [arm64-darwin21]
```
